python/lsst/afw/geom/SphPoint.i

```diff
@@ -0,0 +1,71 @@
+// -*- lsst-c++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+%{
+#include "lsst/afw/geom/SphPoint.h"
+%}
+
+%include "lsst/afw/geom/Angle.h"
+%include "lsst/afw/geom/Point.h"
+
+%useValueEquality(lsst::afw::geom::SphPoint);
+%rename(__getitem__) lsst::afw::geom::SphPoint::operator[];
+
+%include "lsst/afw/geom/SphPoint.h"
+
+// add __str__ and __repr__ methods
+%extend lsst::afw::geom::SphPoint {
+    std::string __str__() const {
+        std::ostringstream os;
+        // Ensure string gives enough precision to enable same-field comparisons
+        os.precision(6);
+        os << std::fixed << (*self);
```

Comment:
I'm not thrilled about forcing a specific precision on the user, though; unless we pick an extreme number of decimal places, there will always be cases where the user wants more.

How about guaranteeing output to at least N decimal places (6?), while allowing the user to request more using the stream precision and/or `boost::format`? That would make the implementation of `<<` a bit more complex, but it would address both our concerns.